### PR TITLE
Enable template rename in read_gds

### PR DESCRIPTION
**Pull request by yoshi74ls181**
Keyword argument `rename` in `read_gds` is a dictionary and is useless for reading a file with unknown cell names.
The new keyword argument `prefix` enables prefixing to all cell names.

### CODE REVIEW

**Comment by heitzmann** [comment]
@yoshi74ls181 
It's a good idea, but why not a suffix? I believe a more general solution would be a template rename:

```python
def read_gds(..., template='{name}', ...):
    ...
    if record[1] in rename:
        name = rename[record[1]]
    else:
        name = template.format(name=record[1])
    ...
```

In that case you could do prefix (`template='prefix-{name}'`), suffix, etc.
An even more general, but possibly more annoying, would be passing a rename function:

```python
def read_gds(..., rename={}, ...):
    ...
    if hasattr(rename, '__call__'):
        name = rename(record[1])
    else:
        name = rename.get(record[1], record[1])
    ...
```

In this case anything is possible. Prefixing would simply be `rename=lambda n: 'prefix-' + n`.

**Comment by yoshi74ls181** [comment]
Thank you for your suggestions.
I especially like the idea of template rename, which I adopted instead of the original prefix functionality.

File: gdspy/__init__.py [hunk 1]
```diff
@@ -3411,7 +3411,7 @@ def read_gds(self,
                  layers={},
                  datatypes={},
                  texttypes={},
-                 prefix=''):
+                 rename_template='{name}'):
```
Comment (heitzmann) [inline gdspy/__init__.py]: `layer`, `datatype` and `texttype` should be the last kwargs. Include this after `rename` — the tests can be modified to follow the new arguments.

File: gdspy/__init__.py [hunk 2]
```diff
@@ -3439,6 +3440,10 @@ def read_gds(self,
         texttypes : dictionary
             Dictionary used to convert the text types in the imported
             cells.  Keys and values must be integers.
+        rename_template : string
+            Template string used to rename the imported cells. Appiled
+            only if the cell name is not in the ``rename`` dictionary.
+            Examples: ``'prefix-{name}'``, ``'{name}-postfix'``
```
Comment (heitzmann) [inline gdspy/__init__.py]: The correct term is *suffix*, not *postfix*.

**Comment by heitzmann** [comment]
@yoshi74ls181 Could you make the pull-request against the branch:next?

**Comment by yoshi74ls181** [comment]
Done.

**Pull request merged**